runtime/darwinia/src/pallets/fee_market.rs

```diff
@@ -0,0 +1,58 @@
+// --- core ---
+use core::cmp;
+// --- substrate ---
+use frame_support::{traits::LockIdentifier, PalletId};
+use sp_runtime::{traits::UniqueSaturatedInto, Permill};
+// --- darwinia ---
+use crate::*;
+use darwinia_fee_market::{Config, RingBalance, Slasher};
+
+frame_support::parameter_types! {
+	pub const FeeMarketPalletId: PalletId = PalletId(*b"da/feemk");
+	pub const TreasuryPalletId: PalletId = PalletId(*b"da/trsry");
+	pub const FeeMarketLockId: LockIdentifier = *b"da/feelf";
+
+	pub const AssignedRelayersNumber: u64 = 3;
+	pub const MinimumRelayFee: Balance = 15 * COIN;
+	pub const CollateralPerOrder: Balance = 100 * COIN;
+	pub const SlashPerBlock: Balance = 2 * COIN;
+	pub const Slot: BlockNumber = 50;
+
+	pub const AssignedRelayersRewardRatio: Permill = Permill::from_percent(60);
+	pub const MessageRelayersRewardRatio: Permill = Permill::from_percent(80);
+	pub const ConfirmRelayersRewardRatio: Permill = Permill::from_percent(20);
+}
+
+impl Config for Runtime {
+	type PalletId = FeeMarketPalletId;
+	type TreasuryPalletId = TreasuryPalletId;
+	type LockId = FeeMarketLockId;
+
+	type AssignedRelayersNumber = AssignedRelayersNumber;
+	type MinimumRelayFee = MinimumRelayFee;
+	type CollateralPerOrder = CollateralPerOrder;
+	type Slot = Slot;
+
+	type AssignedRelayersRewardRatio = AssignedRelayersRewardRatio;
+	type MessageRelayersRewardRatio = MessageRelayersRewardRatio;
+	type ConfirmRelayersRewardRatio = ConfirmRelayersRewardRatio;
+
+	type Slasher = FeeMarketSlasher;
+	type RingCurrency = Ring;
+	type Event = Event;
+	type WeightInfo = ();
+}
+
+pub struct FeeMarketSlasher;
+impl<T: Config> Slasher<T> for FeeMarketSlasher {
+	fn slash(locked_collateral: RingBalance<T>, timeout: T::BlockNumber) -> RingBalance<T> {
+		let slash_each_block = 2 * COIN;
```

Comment:
Request document

Reply:
The docs:  For each block delayed, the order's assigned relayers collateral slashed 2 coin, with a maximum slash value order's locked_collateral.

@AurevoirXavier

Reply:
This should be added to Darwinia Common IMO.

Reply:
The Slasher trait.

Reply:
> This should be added to Darwinia Common IMO.

Different runtime could have different slasher impl and algorithm. The requested doc is for the implementation

Reply:
The comment I wrote above is to explain the slasher implementation, not the trait itself.
> This should be added to Darwinia Common IMO.
Yes, It should be updated to the pangolin/pangoro runtime config as well. We can update this in common later.